plugins/base/src/main/kotlin/transformers/pages/annotations/SinceKotlinTransformer.kt

```diff
@@ -1,93 +1,173 @@
 package org.jetbrains.dokka.base.transformers.pages.annotations
 
+import com.intellij.util.containers.ComparatorUtil.max
 import org.intellij.markdown.MarkdownElementTypes
-import org.jetbrains.dokka.links.DRI
+import org.jetbrains.dokka.DokkaConfiguration
+import org.jetbrains.dokka.Platform
+import org.jetbrains.dokka.base.signatures.KotlinSignatureUtils.annotations
 import org.jetbrains.dokka.model.*
 import org.jetbrains.dokka.model.doc.CustomDocTag
 import org.jetbrains.dokka.model.doc.CustomTagWrapper
+import org.jetbrains.dokka.model.doc.DocumentationNode
 import org.jetbrains.dokka.model.doc.Text
-import org.jetbrains.dokka.model.properties.WithExtraProperties
 import org.jetbrains.dokka.plugability.DokkaContext
 import org.jetbrains.dokka.transformers.documentation.DocumentableTransformer
+import org.jetbrains.dokka.utilities.associateWithNotNull
 import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
 class SinceKotlinTransformer(val context: DokkaContext) : DocumentableTransformer {
+    class Version constructor(str: String) : Comparable<Version> {
+        private val parts: List<Int> = str.split(".").map { it.toInt() }
+
+        override fun compareTo(other: Version): Int {
+            val i1 = parts.listIterator()
+            val i2 = other.parts.listIterator()
+
+            while (i1.hasNext() || i2.hasNext()) {
+                val diff = (if (i1.hasNext()) i1.next() else 0) - (if (i2.hasNext()) i2.next() else 0)
+                if (diff != 0) return diff
+            }
+
+            return 0
+        }
+
+        override fun toString(): String = parts.joinToString(".")
+    }
+
+    private val minSinceKotlinOfPlatform = mapOf(
```

Comment:
`SinceKotlin` detail looks redundant in this case

```suggestion
    private val minVersionOfPlatform = mapOf(
```

Reply:
I would left it, naming costs nothing. `Version` can be confusing. A version of what? Library!? Dokka!? Kotlin?!

Reply:
> A version of a platform?

It's a **min** version of a platform, yeah. As in the "first version in which this platform appeared" - as far as I understand, that's what this property is used for. `common` first appeared in 1.2, `native` was announced in 1.3, `js` in 1.1 (I looked through the "What's changed" archive)

The reason I suggested to remove `SinceKotlin` from it is because it makes you think if `SinceKotlin` is an exceptional case or it has some meaning, like maybe native target was actually introduced in Kotlin 1.1, but min version of `Since Kotlin` of native target is for some reason 1.3. Whereas, as I understand it, `SinceKotlin` adds no explanation to the used version

So the less confusing words that make you question things, the better, imo